py/dml/expr.py

```diff
@@ -80,8 +83,28 @@ def coverity_markers(markers, site=None):
         site_linemark_nocoverity(site)
 
 
+# Allow linemarks to be generated if dml.globals.linemarks_enabled is True
+# and the current output is FileOutput
+# This context manager generates a line directive reset when left unless
+# linemarks were already allowed, or manual_reset is True
+@contextmanager
+def allow_linemarks(manual_reset=False):
```

Comment:
Can we find a better word than `allow` here? to me, 'allow' something indicates that it's an error to do that otherwise. I can understand your reasoning that 'if linemarks are enabled, they will be output unless disallowed", but it's really a symmetric conjunction (linemarks happen whenever they are both "enabled" and "allowed"), so you could just as well swap the two words.

(though I realize I'm starting a discussion on the naming of an internal identifier, so feel free to close the thread with no comment)

Reply:
i have no idea how to best express this concept. I found "allow" and decided that were good enough.